src/PureScript/Backend/Chez/Convert.purs

```diff
@@ -22,28 +24,106 @@ type CodegenEnv =
   { currentModule :: ModuleName
   }
 
-codegenModule :: BackendModule -> ChezExpr
+codegenModule :: BackendModule -> ChezLibrary
 codegenModule { name, bindings, exports } =
   let
     codegenEnv :: CodegenEnv
     codegenEnv = { currentModule: name }
 
-    exports' :: Array Ident
-    exports' = Array.fromFoldable exports
+    exports' :: Array ChezExport
+    exports' = map ExportIdentifier $ coerce $ Array.fromFoldable exports
 
-    bindings' :: Array ChezExpr
-    bindings' = Array.concat $ codegenTopLevelBindingGroup codegenEnv <$> bindings
+    definitions :: NonEmptyArray ChezDefinition
+    definitions = unsafePartial $ fromJust $ NEA.fromArray $ Array.concat $
+      codegenTopLevelBindingGroup codegenEnv <$> bindings
```

Comment:
Given an empty module, this crashes:
```purescript
module Snapshot.Empty where

```

Also caught this on https://github.com/purescm/purescm/pull/76#issuecomment-1473872909 for some reason, but that could just be an oddity with build caching.

There's two ways to go about this:
1. Create a placeholder `define` such that `exprs` can still be inserted
2. Drop the `exprs` field in the `body` and revert to a regular `Array`

PureScript doesn't have expressions at the top-level, so 2. seems the most reasonable for me...

Reply:
The code above is just wrong. The spec says the `libraryBody` is
`〈definition〉 ... 〈expression〉 ...`, which means both `definition` and `expression` can both be empty.

I think I thought that was weird because why have a library that doesn't have anything inside of it? On a second reading, that also means the `export` and `import` lists can be empty. But again, why have a library if it doesn't export anything?

I guess my understanding is just flawed because libraries can also execute code found in the library body's `exprs`.

Reply:
Fixed.

Reply:
> There's two ways to go about this:
> 1. Create a placeholder define such that exprs can still be inserted
> 2. Drop the exprs field in the body and revert to a regular Array

I think we should keep `exprs` because we can use the same type when parsing FFI files.

Reply:
Would it be appropriate to encode the invariant that `exprs` can only be non-empty if `definitions` is non-empty as well?

Reply:
This is weird... Running the following in the REPL works:
```scheme
(library (a) (export) (import))
```